src/server/compositor/multi_threaded_compositor.cpp

```diff
@@ -254,18 +210,15 @@ class CompositingFunctor
     std::shared_ptr<mc::DisplayBufferCompositorFactory> const compositor_factory;
     mg::DisplaySyncGroup& group;
     std::shared_ptr<mc::Scene> const scene;
+    std::binary_semaphore wakeup;
     bool running;
```

Comment:
We're no longer guarding `running` with `run_mutex`. So we should make it `atomic<>`

Reply:
I'm *moderately* certain that the semaphore provides a sufficient barrier, but just using an `atomic<bool>` will definitely make it *obviously* correct.

Reply:
I don't see that there was a sufficient barrier:

Thread A: `CompositingFunctor::operator()()` - acquires
Thread B: `CompositingFunctor::stop()` - running = false
Thread A: `CompositingFunctor::operator()()` - if (running) // undefined behavour: no guarantee running is one of true or false (even though it is on Intel, for example)